tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -5773,106 +5756,128 @@ Value *CGMSHLSLRuntime::EmitHLSLInitListExpr(CodeGenFunction &CGF, InitListExpr
   }
 }
 
-static void FlatConstToList(Constant *C, SmallVector<Constant *, 4> &EltValList,
-                            QualType Type, CodeGenTypes &Types,
-                            bool bDefaultRowMajor) {
+static void FlatConstToList(CodeGenTypes &Types, bool bDefaultRowMajor,
```

Comment:
This method is partially rewritten to make it output the QualTypes of the constant elements, though I also took the opportunity to clean things up a bit.